src/main/java/org/sonarlint/intellij/trigger/SonarDocumentListener.java

```diff
@@ -52,14 +55,17 @@
   private final Map<VirtualFile, Long> eventMap;
   private final EventWatcher watcher;
   private final int timerMs;
+  private final OpenFileChecker openFileChecker;
 
   public SonarDocumentListener(Project project, SonarLintGlobalSettings globalSettings, SonarLintSubmitter submitter,
     EditorFactory editorFactory, SonarLintAppUtils utils, FileDocumentManager docManager) {
-    this(project, globalSettings, submitter, editorFactory, utils, docManager, DEFAULT_TIMER_MS);
+    this(project, globalSettings, submitter, editorFactory, utils, docManager, DEFAULT_TIMER_MS,
+            new DefaultOpenFileChecker(project));
   }
 
   public SonarDocumentListener(Project project, SonarLintGlobalSettings globalSettings, SonarLintSubmitter submitter,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS00107)